commands/pipelines/diff.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+let cli          = require('heroku-cli-util');
+let co           = require('co');
+let bluebird     = require('bluebird');
+let request      = bluebird.promisify(require('request'));
+let _            = require('lodash');
+
+const PROMOTION_ORDER = ['development', 'staging', 'production'];
+const V3_HEADER = 'application/vnd.heroku+json; version=3';
+const PIPELINES_HEADER = V3_HEADER + '.pipelines';
+
+// Helper functions
+
+function kolkrabbiRequest(url, token) {
+  return request({
+    method: 'GET',
+    url: url,
+    headers: {
+      authorization: 'Bearer ' + token
+    },
+    json: true
+  }).spread(function (res, body) {
+    if (res.statusCode === 404) {
+      let err = new Error('404');
+      err.name = 'NOT_FOUND';
+      throw err;
+    } else if (res.statusCode >= 400) {
+      // TODO: This could potentially catch some 4xx errors that we might want to handle with a
+      // specific error message.
+      throw new Error('failed to fetch diff because of an internal server error.');
+    }
+    return body;
+  });
+}
+
+function* getLatestCommitHash(heroku, appName, appId) {
+  const release = yield heroku.request({
+    method: 'GET',
+    path: `/apps/${appId}/releases`,
+    headers: { 'Accept': V3_HEADER, 'Range': 'version ..; order=desc,max=1' },
+    partial: true
+  });
+  const slug = yield heroku.request({
+    method: 'GET',
+    path: `/apps/${appId}/slugs/${release[0].slug.id}`,
+    headers: { 'Accept': V3_HEADER }
+  });
+  return { name: appName, hash: slug.commit };
+}
+
+function* diff(sourceApp, downstreamApp, repo, githubToken, herokuUserAgent) {
+  if (sourceApp.hash === downstreamApp.hash) {
+    console.log(`\neverything is up to date between ${sourceApp.name} and ${downstreamApp.name}`);
+    return;
+  }
+  const githubDiff = yield request({
+    url: `https://api.github.com/repos/${repo}/compare/${downstreamApp.hash}...${sourceApp.hash}`,
+    headers: {
+      authorization: 'token ' + githubToken,
+      'user-agent': herokuUserAgent,
+    },
+    json: true
+  }).get(1);
+
+  console.log(`\n${sourceApp.name} is ahead of ${downstreamApp.name} by ${githubDiff.ahead_by} commit${githubDiff.ahead_by === 1 ? '' : 's'}:`);
+  for (let i = githubDiff.commits.length - 1; i >= 0; i--) {
+    let commit = githubDiff.commits[i];
+    let abbreviatedHash = commit.sha.substring(0, 7);
+    let authoredDate = commit.commit.author.date;
+    let authorName = commit.commit.author.name;
+    let message = commit.commit.message.split('\n')[0];
+    console.log(`  ${abbreviatedHash}  ${authoredDate}  ${message} (${authorName})`);
+  }
+}
+
+function* fetchPipelineCoupling(heroku, appName) {
+  try {
+    return yield heroku.request({
+      method: 'GET',
+      path: `/apps/${appName}/pipeline-couplings`,
+      headers: { 'Accept': PIPELINES_HEADER }
+    });
+  } catch (err) {
+    console.log(err);
```

Comment:
You could change this to a `cli.debug` statement. That way it'll be hidden for most users.